carapace-server/src/test/java/org/carapaceproxy/cluster/impl/ZookKeeperACLTest.java

```diff
@@ -0,0 +1,177 @@
+
+    @ClassRule
+    public static TemporaryFolder folder = new TemporaryFolder();
+
+    @Rule
+    public TemporaryFolder tmpDir = new TemporaryFolder();
+    String peerId1 = "p1";
+    String peerId2 = "p2";
+    String peerId3 = "p3";
+
+    /**
+     *
+     */
+    @BeforeClass
+    public static void setUpEnvironment() {
+        File file = new File("src/test/resources/jaas/test_jaas.conf");
+        System.setProperty("java.security.auth.login.config", file.getAbsolutePath());
+        assertTrue(file.isFile());
+        Configuration.getConfiguration().refresh();
+    }
+
+    /**
+     *
+     * @throws InterruptedException
+     * @throws IOException
+     */
+    @AfterClass
+    public static void cleanUpEnvironment() throws InterruptedException, IOException {
+        System.clearProperty("java.security.auth.login.config");
+        Configuration.getConfiguration().refresh();
+    }
+
+    @Test
+    public void testUseAcl() throws Exception {
+        Map<String, Object> customProperties = new HashMap<>();
+        customProperties.put("authProvider.1", "org.apache.zookeeper.server.auth.SASLAuthenticationProvider");
+        InstanceSpec def = InstanceSpec.newInstanceSpec();
+        InstanceSpec ss = new InstanceSpec(folder.newFolder(), def.getPort(), def.getElectionPort(),
+                def.getQuorumPort(), false /*deleteDataDirectoryOnClose*/, def.getServerId(),
+                def.getTickTime(), def.getMaxClientCnxns(), customProperties, def.getHostname());
+
+        try (TestingServer testingServer = new TestingServer(ss,
+                false);) {
+            testingServer.start();
+            try (ZooKeeperGroupMembershipHandler peer1 = new ZooKeeperGroupMembershipHandler(testingServer.getConnectString(),
+                    6000, true /*acl */, peerId1, Collections.EMPTY_MAP, new Properties());
+                    ZooKeeperGroupMembershipHandler peer2 = new ZooKeeperGroupMembershipHandler(testingServer.getConnectString(),
+                            6000, true /*acl */, peerId2, Collections.EMPTY_MAP, new Properties())) {
+                peer1.start();
+                peer2.start();
+                List<String> peersFrom1 = peer1.getPeers();
+                List<String> peersFrom2 = peer2.getPeers();
+                assertEquals(Arrays.asList(peerId1, peerId2), peersFrom1);
+                assertEquals(Arrays.asList(peerId1, peerId2), peersFrom2);
+
+                AtomicInteger eventFired2 = new AtomicInteger();;
+                peer2.watchEvent("foo", new GroupMembershipHandler.EventCallback() {
+                    @Override
+                    public void eventFired(String eventId) {
+                        eventFired2.incrementAndGet();
+                    }
+
+                    @Override
+                    public void reconnected() {
+
+                    }
+                });
+
+                peer1.fireEvent("foo");
+
+                for (int i = 0; i < 10; i++) {
```

Comment:
Maybe use TestUtils#waitForCondition. It's basically the same thing but we avoi some clutter